azurerm/resource_arm_cognitive_account.go

```diff
@@ -0,0 +1,258 @@
+package azurerm
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/cognitiveservices/mgmt/2017-04-18/cognitiveservices"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmCognitiveAccount() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmCognitiveAccountCreate,
+		Read:   resourceArmCognitiveAccountRead,
+		Update: resourceArmCognitiveAccountUpdate,
+		Delete: resourceArmCognitiveAccountDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"kind": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateCognitiveAccountKind,
+			},
+
+			"sku": {
+				Type:     schema.TypeSet,
```

Comment:
we should be able to make this a `TypeList` rather than a `TypeSet`- which means it can be expanded via:

```
inputs := d.Get("sku").([]interface{})
input := inputs[0].(map[string]interface{})
# key into input to get the values
```

and flattened via:

```
func flatten... (input *cognitiveservuces.Sku)[]interface{} {
  output := make(map[string]interface{}, 0)
  output["foo"] = input.Bar
  return []interface{}{output}
}
```